src/taskgraph/decision.py

```diff
@@ -166,6 +167,9 @@ def get_decision_parameters(graph_config, options):
     except UnicodeDecodeError:
         commit_message = ""
 
+    if not options.get("base_ref"):
+        parameters["base_ref"] = repo.default_branch
```

Comment:
Why is `default_branch` needed if we have `base_rev`? Wouldn't the latter be a better default anyway?

Reply:
Thanks for the explanation!

I'm not sure if this belongs in documentation, but I might be inclined to include it in the commit message itself!

Reply:
Sounds good! I just reformated my previous comment and put it in https://github.com/taskcluster/taskgraph/pull/93/commits/877c1b8ac499196badd129ae43d79c15aab2c9fd

Reply:
End-to-end testing showed there were a few things wrong in my previous explanation: we can't always rely on the `base_ref`. 

## Same head branch name, but force-pushed commits

Let's imagine we force-push several commits commit. The `base_ref` remains the same as the `head_ref` one. If we rely on the `base_ref`, then the `merge-base` between `base_ref` and the `head_rev` yields a single commit, the `head_rev`. This is a problem since we force pushed several commits.


## New head branch, but same code

This case needs to be split into 2. If the new branch is 100% the same as the old one, then `base_ref` is still the right way to go. 
If the head branch is slightly different (say a commit was amended), Github doesn't provides a wrong `base_ref` and gives `NULL_REVISION` as the `base_rev`. In this case, we're overriding the `base_ref` to match the default branch. It a saner default than keeping the provided `base_ref` which behaves as a single commit was pushed (like in the force-push` scenario.

## Changes

That's why I changed `run-task` to fetch the `base_ref` and the `base_rev` if it's the `NULL_REVISION`, if they're provided. The heuristic to determine more accurate `base_ref`/`base_rev` we changed too.